src/database/volatile/mod.rs

```diff
@@ -55,3 +67,68 @@ impl Tabular for Database<'_> {
         todo!()
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    /// A maximally simple Record implementation, consisting of only a BitVec value
+    /// Created for test purposes
+    pub struct Rec {
+        value: BitVec<u8, Msb0>,
+    }
+
+    impl Rec {
+        pub fn initialize(val: BitVec<u8, Msb0>) -> Self {
+            Self { value: val.clone() }
+        }
+    }
+
+    impl Record for Rec {
+        fn raw(&self) -> &BitSlice<u8, Msb0> {
+            return &self.value[..];
+        }
+    }
+
+    /// This test
```

Comment:
Thanks for writing test docstrings! 

These of course don't need to be "client oriented" like the `Database` docstring above, but let's keep all these grammatically correct:

```
This test:
- Creates... state `test_state` and `Record` `test_rec`. 
- ...
```

Note that we also get markdown syntax in docstrings, hence the backticks.